src/utils/externalLinkHelper.js

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+
+const domain = `https://rpf.io/`;
+const rpfDomain = new RegExp(`^${domain}`);
+const allowedInternalLinks = [new RegExp(`^#[a-zA-Z0-9]+`)];
+const allowedExternalLinks = [rpfDomain];
+
+const useExternalLinkState = () => {
+  const [externalLink, setExternalLink] = useState();
+
+  const handleAllowedExternalLink = (linkTo, dispatch, triggerCodeRun) => {
+    setExternalLink(linkTo);
+    dispatch(triggerCodeRun());
+  };
+
+  const handleRegularExternalLink = (linkTo, setPreviewFile) => {
+    setExternalLink(null);
+    setPreviewFile(`${linkTo}.html`);
+    dispatch(triggerCodeRun());
+  };
```

Comment:
☝️ these are because if they're removed at this level you can define and pass them in a level higher just the once [here](https://github.com/RaspberryPiFoundation/editor-ui/pull/876/files#r1450079169) 👍